### PR TITLE
DDF-2108 Remove dependencies on and disable the Felix console

### PR DESCRIPTION
**Pull request by jckilmer**
#### What does this PR do?
Disables the Felix webconsole as well as removes dependencies on it. Those dependencies included some branding capability which had to be refactored since other modules depend on it (landing page, platform ui config, etc.).

#### Who is reviewing it (please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)?
@tbatie 
@coyotesqrl 
@ryeats 

#### Choose 2 committers to review/merge the PR (please choose ONLY two committers from below, delete the rest).
@pklinef
@stustison

#### How should this be tested?
Full build including itests; deploy and verify the felix web console is disabled. Verify branding on pages (admin ui, search ui, landing page) as appropriate. Also verify landing page and platform ui configurations can be changed and are updated correctly.

#### Any background context you want to provide?
N/A

#### What are the relevant tickets?
DDF-2108

#### Screenshots (if appropriate)
N/A

#### Checklist:
- [x] Documentation Updated
- [x] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

### CODE REVIEW

File: platform/admin/ui/pom.xml [hunk 1]
```diff
@@ -66,6 +68,10 @@
             <type>tar.gz</type>
             <classifier>npm</classifier>
         </dependency>
+        <dependency>
```
Comment (coyotesqrl) [inline platform/admin/ui/pom.xml]: With this added here, do you still need the `org.apache.karaf.webconsole.console` dependency in this pom?

File: distribution/branding-plugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml [hunk 2]
```diff
@@ -9,13 +10,13 @@
...
  * <http://www.gnu.org/licenses/lgpl.html>.
  *
- **/
+**/
 -->
-<blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0"
-        >
 
-    <bean id="provider" class="org.codice.ddf.branding.impl.BrandingResourceProviderImpl"/>
+<blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0">
```
Comment (coyotesqrl) [inline distribution/branding-plugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: Missing namespaces.

File: platform/metrics/platform-metrics-webconsole-plugin/pom.xml [hunk 3]
```diff
@@ -1,141 +0,0 @@
-<?xml version="1.0" encoding="UTF-8" standalone="no"?>
```
Comment (coyotesqrl) [inline platform/metrics/platform-metrics-webconsole-plugin/pom.xml]: Do metrics work on the admin console without this plugin? Was it specific to the Felix webconsole?

File: platform/platform-configuration/pom.xml [hunk 4]
```diff
@@ -80,9 +82,10 @@
                 <configuration>
                     <instructions>
                         <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
-                        <Embed-Dependency>json-smart,common-system,org.apache.felix.utils,commons-lang3</Embed-Dependency>
+                        <Embed-Dependency>
+                            json-smart,common-system,org.apache.felix.utils,commons-lang3
+                        </Embed-Dependency>
                         <Import-Package>
```
Comment (coyotesqrl) [inline platform/platform-configuration/pom.xml]: You can remove the entire `<Import-Package>` directive now as `*` is the default.

File: platform/admin/ui/pom.xml (hunk 1 above)
Comment (jckilmer) [inline platform/admin/ui/pom.xml]: No I don't think so, actually I think `org.apache.karaf.webconsole.console` can be removed from several poms.

File: platform/metrics/platform-metrics-webconsole-plugin/pom.xml (hunk 3 above)
Comment (jckilmer) [inline platform/metrics/platform-metrics-webconsole-plugin/pom.xml]: I believe it was specific to the Felix webconsole, I performed some ingests and queries and they showed up on the admin ui metrics page.

File: distribution/branding-plugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml (hunk 2 above)
Comment (jckilmer) [inline distribution/branding-plugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: Is that namespace not sufficient?

Comment (coyotesqrl) [inline distribution/branding-plugin/src/main/resources/OSGI-INF/blueprint/blueprint.xml]: It is, in fact. But I know the odds are that someone will copy/paste your blueprint at some point and if they need other blueprint functionality will be missing it. It's less about need for today than future-proofing newbie mistakes.

File: platform/admin/ui/pom.xml (hunk 1 above)
Comment (jckilmer) [inline platform/admin/ui/pom.xml]: Fixed.

File: platform/platform-configuration/pom.xml (hunk 4 above)
Comment (jckilmer) [inline platform/platform-configuration/pom.xml]: Fixed.

File: catalog/ui/search-ui/simple/src/main/java/org/codice/ddf/ui/searchui/simple/properties/ConfigurationStore.java [hunk 5]
```diff
@@ -15,13 +15,12 @@
 package org.codice.ddf.ui.searchui.simple.properties;
 
 import org.apache.commons.lang.StringUtils;
-import org.apache.felix.webconsole.BrandingPlugin;
+import org.codice.ddf.branding.BrandingPlugin;
 
 /**
  * Stores external configuration properties.
  *
  * @author ddf.isgs@lmco.com
```
Comment (coyotesqrl) [inline catalog/ui/search-ui/simple/src/main/java/org/codice/ddf/ui/searchui/simple/properties/ConfigurationStore.java]: Remove author comment.

File: distribution/branding-api/pom.xml [hunk 6]
```diff
@@ -1,47 +1,45 @@
...
-        DDF branding for the DDF Web Console. This fragment
-        attaches to the org.apache.felix.webconsole bundle to provide branding
-        through the /META-INF/webconsole.properties.
-  </description>
     <build>
         <plugins>
             <plugin>
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
-                <extensions>true</extensions>
```
Comment (coyotesqrl) [inline distribution/branding-api/pom.xml]: I think this still needs to be here.

File: distribution/ddf-branding-plugin/pom.xml [hunk 7]
```diff
@@ -10,50 +10,73 @@
...
-            <resource>
-                <directory>src/main/resources</directory>
-                <filtering>true</filtering>
-            </resource>
-        </resources>
         <plugins>
             <plugin>
                 <groupId>org.apache.felix</groupId>
                 <artifactId>maven-bundle-plugin</artifactId>
-                <extensions>true</extensions>
```
Comment (coyotesqrl) [inline distribution/ddf-branding-plugin/pom.xml]: I think this needs to be restored.

File: distribution/ddf-branding-plugin/pom.xml [hunk 8]
```diff
@@ -10,50 +10,73 @@
...
                 <configuration>
                     <instructions>
                         <Bundle-SymbolicName>${project.artifactId}</Bundle-SymbolicName>
-                        <Export-Package/>
+                        <Bundle-Name>${project.name}</Bundle-Name>
+                        <Private-Package>
```
Comment (coyotesqrl) [inline distribution/ddf-branding-plugin/pom.xml]: I think you can remove this and restore the `<Export-Package/>`. That will export nothing as well.

File: distribution/docs/src/main/resources/_contents/_spatial-contents/managing-spatial-contents.adoc [hunk 9]
```diff
@@ -123,8 +123,8 @@ geonames:update
...
 
-The index location can be configured via the Admin UI or the Felix Web Console. By default, the index location is `data/geonames-index`. If you specify a relative path, it is relative to the location of the unzipped
+The index location can be configured via the Admin UI. By default, the index location is `data/geonames-index`. If you specify a relative path, it is relative to the location of the unzipped
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_spatial-contents/managing-spatial-contents.adoc]: Can you change Admin UI to Admin Console (or better yet: ${admin-console}?

File: distribution/docs/src/main/resources/_contents/_spatial-contents/managing-spatial-contents.adoc [hunk 10]
```diff
@@ -123,8 +123,8 @@ geonames:update
...
 ${branding} distribution. You may specify an absolute path if you want the index to be located somewhere else.
 
-The `-c` or `--create` flag can be added to create a new GeoNames index. This will overwrite any existing index at the location specified in the Admin UI or Felix Web Console. The new index will be filled with the entries in the file you pass to the command. You must create an index before you can add additional entries to it (i.e. running the command without the `-c` or `--create` flag).
+The `-c` or `--create` flag can be added to create a new GeoNames index. This will overwrite any existing index at the location specified in the Admin UI. The new index will be filled with the entries in the file you pass to the command. You must create an index before you can add additional entries to it (i.e. running the command without the `-c` or `--create` flag).
```
Comment (ricklarsen) [inline distribution/docs/src/main/resources/_contents/_spatial-contents/managing-spatial-contents.adoc]: And here to Admin Console

**Comment by coyotesqrl** [comment]
👍

File: catalog/ui/search-ui/simple/src/main/java/org/codice/ddf/ui/searchui/simple/properties/ConfigurationStore.java (hunk 5 above)
Comment (jckilmer) [inline catalog/ui/search-ui/simple/src/main/java/org/codice/ddf/ui/searchui/simple/properties/ConfigurationStore.java]: Fixed.

File: distribution/branding-api/pom.xml (hunk 6 above)
Comment (jckilmer) [inline distribution/branding-api/pom.xml]: Fixed.

File: distribution/ddf-branding-plugin/pom.xml (hunk 7 above)
Comment (jckilmer) [inline distribution/ddf-branding-plugin/pom.xml]: Fixed.

File: distribution/ddf-branding-plugin/pom.xml (hunk 8 above)
Comment (jckilmer) [inline distribution/ddf-branding-plugin/pom.xml]: Fixed.

File: distribution/docs/src/main/resources/_contents/_spatial-contents/managing-spatial-contents.adoc (hunk 9 above)
Comment (jckilmer) [inline distribution/docs/src/main/resources/_contents/_spatial-contents/managing-spatial-contents.adoc]: Fixed.

File: distribution/docs/src/main/resources/_contents/_spatial-contents/managing-spatial-contents.adoc (hunk 10 above)
Comment (jckilmer) [inline distribution/docs/src/main/resources/_contents/_spatial-contents/managing-spatial-contents.adoc]: Fixed.

File: distribution/branding-api/src/main/java/org/codice/ddf/branding/BrandingPlugin.java [hunk 11]
```diff
@@ -10,20 +10,31 @@
...
-    public byte[] getResourceAsBytes(String path) throws IOException;
+public interface BrandingPlugin {
+
+    String getProductName();
+
+    String getProductURL();
+
+    String getProductImage();
+
+    String getProductImageAsBytes() throws IOException;
```
Comment (pklinef) [inline distribution/branding-api/src/main/java/org/codice/ddf/branding/BrandingPlugin.java]: The AsBytes with a return type of String seems odd.  Maybe `getBase64ProductImage` instead?

File: distribution/ddf-branding-plugin/src/main/java/org/codice/ddf/branding/impl/DdfBrandingPlugin.java [hunk 12]
```diff
@@ -0,0 +1,147 @@
...
+import org.apache.commons.io.IOUtils;
+import org.codice.ddf.branding.BrandingPlugin;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class DdfBrandingPlugin implements BrandingPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DdfBrandingPlugin.class);
+
+    private static String brandingProperties = "/META-INF/branding.properties";
```
Comment (pklinef) [inline distribution/ddf-branding-plugin/src/main/java/org/codice/ddf/branding/impl/DdfBrandingPlugin.java]: I would suggest making this non-static, replacing its setter with a constructor that takes the value, and rename this field to something that make it clearer that this is the branding property file path.

File: distribution/branding-api/src/main/java/org/codice/ddf/branding/BrandingPlugin.java (hunk 11 above)
Comment (pklinef) [inline distribution/branding-api/src/main/java/org/codice/ddf/branding/BrandingPlugin.java]: Same comment for vendor and favicon base64 methods.

**Comment by pklinef** [comment]
:+1: after addressing my comments.

File: distribution/branding-api/src/main/java/org/codice/ddf/branding/BrandingPlugin.java (hunk 11 above)
Comment (jckilmer) [inline distribution/branding-api/src/main/java/org/codice/ddf/branding/BrandingPlugin.java]: Fixed.

File: distribution/ddf-branding-plugin/src/main/java/org/codice/ddf/branding/impl/DdfBrandingPlugin.java (hunk 12 above)
Comment (jckilmer) [inline distribution/ddf-branding-plugin/src/main/java/org/codice/ddf/branding/impl/DdfBrandingPlugin.java]: Fixed.

**Pull request merged**